astropy/modeling/rotations.py

```diff
@@ -187,9 +187,6 @@ def evaluate(cls, x, y, angle):
         single coordinate pair.
         """
 
-        if x.shape != y.shape:
-            raise ValueError("Expected input arrays to have the same shape")
```

Comment:
In other words, there's this sort of implicit assumption, which really ought to be handled better, that the `evaluate` methods return an array of shape at least `(1,)`.  That it never returns array scalars of shape `()`.